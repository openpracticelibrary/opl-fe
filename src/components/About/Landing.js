import React from 'react';
import { makeStyles } from '@material-ui/core/styles/index';
import Box from '@material-ui/core/Box';
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import LoginButton from "../shared/Login/LoginButton";
import GrayDiagonals from "../../assets/icons/graylines.svg";
import ExploreStickies from "../../assets/icons/explorestickies.png";

const useStyles = makeStyles((theme) => ({
  titleBox: {
    display: "flex",
    flexDirection: "row",
    alignContent: "center",
    alignItems: "center",
    backgroundImage: `linear-gradient(180.18deg, #F3F3F3 0%, #FFFFFF 13.64%, #FFFFFF 43.94%, rgba(255,255,255,0.94) 79.58%, rgba(255,255,255,0.62) 100%), url(${GrayDiagonals})`,
    height: theme.spacing(55),
    width: "100%",
  },
  loginBox: {
    position: "absolute",
    top: 0,
    right: 0,
    padding: theme.spacing(3),
  },
  textBox: {
    display: "flex",
    flexWrap: "wrap",
    justifyContent: "left",
    width: "75%",
  },  
  hello: {
    paddingBottom: theme.spacing(5),
  },
  welcome: {
    paddingBottom: theme.spacing(1),
  },
}));

const Landing = (props) => {
  const classes = useStyles();

  return (
    <>
      <Grid
        container
        direction="row"
        className={classes.titleBox}
      >
        <Grid item xs={1}/>
        <Grid item xs={5}>
          <Box className={classes.textBox}>
            <Typography variant={"h1"} className={classes.hello} data-testid="hellotext">
              {props.greeting}
            </Typography>
            <Typography variant={"h5"} className={classes.welcome} data-testid="welcomeText">
              {props.welcomeMessage}
            </Typography>
          </Box>
        </Grid>
        <Grid item xs={3}>
          <img src={ExploreStickies} alt="About Us" height="auto" width="600px" />
          <Box className={classes.loginBox}>
            <LoginButton
              navigate={props.navigate}
              redirect="/practice"
            />
          </Box>
        </Grid>
      </Grid>
    </>
  )
};

export default Landing;
