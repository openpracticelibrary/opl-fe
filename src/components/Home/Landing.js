import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ArrowDownwardIcon from '@material-ui/icons/ArrowDownward';
import { OPLLogo } from "../../assets/icons";
import LoginButton from "../shared/Login/LoginButton";
import GrayDiagonals from "../../assets/icons/graylines.svg";
import HomeStickies from "../../assets/images/WhiteBoarding.png";

import { Box, Grid, IconButton, Typography } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  titleBox: {
    display: "flex",
    flexDirection: "row",
    alignContent: "center",
    alignItems: "center",
    backgroundImage: `linear-gradient(180.18deg, #F3F3F3 0%, #FFFFFF 13.64%, #FFFFFF 43.94%, rgba(255,255,255,0.94) 79.58%, rgba(255,255,255,0.62) 100%), url(${GrayDiagonals})`,
    height: "910px",
    width: "100%",
  },
  logo: {
    position: "absolute",
    top: 0,
    left: 0,
    marginTop: theme.spacing(-12),
    padding: theme.spacing(3),
  },
  loginBox: {
    position: "absolute",
    top: 0,
    right: 0,
    padding: theme.spacing(3),
  },
  textBox: {
    display: "flex",
    flexWrap: "wrap",
    justifyContent: "left",
    width: "75%",
  },
  hello: {
    paddingBottom: theme.spacing(8),
  },
  welcome: {
    paddingBottom: theme.spacing(8),
  },
  arrowIcon: {
    margin: theme.spacing(2),
    color: theme.palette.common.true_white,
    backgroundColor: theme.palette.common.discovery_blue,
  },
}));

const Landing = (props) => {
  const classes = useStyles();

  const handleClick = () => {
    props.communityRef.current.scrollIntoView({
      behavior: "smooth",
      block: "start",
    });
  };

  return (
    <Grid
      container
      direction="row"
      className={classes.titleBox}
    >
      <Grid item xs={1}>
      </Grid>
      <Grid item xs={3}>
        <Box className={classes.logo}>
          <OPLLogo height="250px" width="383px" />
        </Box>
        <Box className={classes.textBox}>
          <Typography variant={"h1"} className={classes.hello} data-testid="hellotext">
            <b>{props.greeting}</b>
          </Typography>
          <Typography variant={"h5"} className={classes.welcome} data-testid="welcomeText">
            {props.welcomeMessage}
          </Typography>
          <Typography variant={"h5"} className={classes.question} data-testid="inquiryText">
            {props.inquiry}
          </Typography>
          <Typography variant={"button"} component="p">
            <b>Tell me more</b>
            <IconButton className={classes.arrowIcon} onClick={handleClick} data-testid="tellMeMoreButton">
              <ArrowDownwardIcon />
            </IconButton>
          </Typography>
        </Box>
      </Grid>
      <Grid item xs={6}>
        <img src={HomeStickies} alt="Welcome to OPL" height="auto" width="900px" />
        <Box className={classes.loginBox}>
          <LoginButton
            navigate={props.navigate}
            redirect="/practice"
          />
        </Box>
      </Grid>
    </Grid>
  )
};

export default Landing;
