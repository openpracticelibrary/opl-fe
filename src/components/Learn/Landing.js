import React from "react";
import { useTheme } from "@material-ui/core/styles";
import ArrowDownwardIcon from "@material-ui/icons/ArrowDownward";

import LoginButton from "../shared/Login/LoginButton";
import GrayDiagonals from "../../assets/images/graylines.svg";
import HomeStickies from "../../assets/images/WhiteBoarding.png";
import { Box } from "@material-ui/core";
import OplBox from "../shared/components/OplBox";
import OplTypography from "../shared/components/OplTypography";
import OplIconButton from "../shared/components/OplIconButton";

const Landing = props => {
  const theme = useTheme();

  const handleClick = () => {
    props.communityRef.current.scrollIntoView({
      behavior: "smooth",
      block: "start"
    });
  };

  return (
    <OplBox
      display="flex"
      flexDirection="row"
      justifyContent="center"
      alignItems="center"
      height="100vh"
      bg={`linear-gradient(180.18deg, #F3F3F3 0%, #FFFFFF 13.64%, #FFFFFF 33.94%, rgba(255,255,255,0.94) 69.58%, rgba(255,255,255,0.62) 100%), url(${GrayDiagonals})`}
      px={8}
    >
      <Box position="absolute" top={2} left={2} p={3}>
        <OplTypography variant="h6" color="primary">
          Open Practice Library
        </OplTypography>
      </Box>
      <Box
        display="flex"
        flexDirection="column"
        width={{ xs: "100%", md: "25%" }}
        justifyContent="flex-start"
      >
        <OplTypography variant={"h1"} pb={8} data-testid="hellotext">
          {props.greeting}
        </OplTypography>
        <OplTypography variant={"h5"} pb={8} data-testid="welcomeText">
          {props.welcomeMessage}
        </OplTypography>
        <OplTypography variant={"h5"} data-testid="inquiryText">
          {props.inquiry}
        </OplTypography>
        <OplTypography variant={"button"} component="p">
          <b>Tell me more</b>
          <OplIconButton
            m={2}
            iconColor="white"
            hoverColor={theme.palette.secondary.main}
            bgcolor={theme.palette.primary.main}
            onClick={handleClick}
            data-testid="tellMeMoreButton"
          >
            <ArrowDownwardIcon />
          </OplIconButton>
        </OplTypography>
      </Box>
      <img src={HomeStickies} alt="Welcome to OPL" height="auto" width="50%" />
      <LoginButton navigate={props.navigate} redirect="/practice" />
    </OplBox>
  );
};

export default Landing;
