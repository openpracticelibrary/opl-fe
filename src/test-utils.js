import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@chakra-ui/core'
import customTheme from './theme';

const ChakraThemeProvider = ({ children }) => {
  return (
    <ThemeProvider theme={customTheme}>
      {children}
    </ThemeProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: ChakraThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

