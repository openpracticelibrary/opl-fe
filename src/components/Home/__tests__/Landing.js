import React from "react";
import { cleanup, render } from "../../../test-utils";
import "@testing-library/jest-dom/extend-expect";
import { MockedProvider } from "@apollo/react-testing";
import Landing from '../Landing';

afterEach(cleanup);

it("renders the Explore hero without crashing", () => {
  const { getByTestId } = render(
    <MockedProvider mocks={[]}>
      <Landing greeting="Wazzup" welcomeMessage="What's up" inquiry="What is up?" />
    </MockedProvider>
  );

  expect(getByTestId("hellotext")).toHaveTextContent("Wazzup");
  expect(getByTestId("welcomeText")).toHaveTextContent("What's up");
  expect(getByTestId("inquiryText")).toHaveTextContent("What is up?");
});

it("matches the snapshot", () => {
  const { asFragment } = render(
    <MockedProvider mocks={[]}>
      <Landing greeting="Wazzup" welcomeMessage="What's up" inquiry="What is up?" />
    </MockedProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
