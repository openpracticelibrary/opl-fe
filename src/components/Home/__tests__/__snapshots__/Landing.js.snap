// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-titleBox-230 MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
  />
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
  >
    <div
      className="MuiBox-root MuiBox-root-341 makeStyles-logo-231"
    >
      <svg
        height="250px"
        viewBox="0 0 185 18"
        width="383px"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <defs>
          <filter
            filterUnits="objectBoundingBox"
            height="113.8%"
            id="prefix__a"
            width="102.2%"
            x="-.5%"
            y="-3.4%"
          >
            <feOffset
              dx="2"
              dy="2"
              in="SourceAlpha"
              result="shadowOffsetOuter1"
            />
            <feColorMatrix
              in="shadowOffsetOuter1"
              values="0 0 0 0 0.912335824 0 0 0 0 0.912335824 0 0 0 0 0.912335824 0 0 0 1 0"
            />
          </filter>
          <text
            fill="#2E2E2E"
            fontFamily="Chewy-Regular, Chewy"
            fontSize="19"
            fontWeight="normal"
            id="prefix__b"
          >
            <tspan
              x="28"
              y="43"
            >
              OPEN PRACTICE LIBRARY
            </tspan>
          </text>
        </defs>
        <g
          fill="#2E2E2E"
          fillRule="evenodd"
          transform="translate(-28 -30)"
        >
          <use
            filter="url(#prefix__a)"
            xlinkHref="#prefix__b"
          />
          <use
            xlinkHref="#prefix__b"
          />
        </g>
      </svg>
    </div>
    <div
      className="MuiBox-root MuiBox-root-342 makeStyles-textBox-233"
    >
      <h4
        className="MuiTypography-root makeStyles-hello-234 MuiTypography-h4"
        data-testid="hellotext"
      >
        Wazzup
      </h4>
      <h5
        className="MuiTypography-root makeStyles-welcome-235 MuiTypography-h5"
        data-testid="welcomeText"
      >
        What's up
      </h5>
      <h5
        className="MuiTypography-root MuiTypography-h5"
        data-testid="inquiryText"
      >
        What is up?
      </h5>
      <p
        className="MuiTypography-root MuiTypography-button"
      >
        <b>
          Tell me more
        </b>
        <button
          className="MuiButtonBase-root MuiIconButton-root makeStyles-arrowIcon-236"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </button>
      </p>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  >
    <img
      alt="Welcome to OPL"
      height="auto"
      src="homestickies.png"
      width="900px"
    />
    <div
      className="MuiBox-root MuiBox-root-394 makeStyles-loginBox-232"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-loginButton-397"
        data-testid="loginButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root makeStyles-buttonText-398 MuiTypography-button"
          >
            Login
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;
